### ISSUE TITLE
lotus sync stucked with error "reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets"

### ISSUE THREAD

**Issue by ysh0566** [filecoin-project/lotus#7803]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [X] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.13.2+mainnet+git.e05fdf892.dirty+api1.4.0
Local: lotus version 1.13.2+mainnet+git.e05fdf892.dirty
```

### Describe the Bug

Client subscribe mpool with MpoolSub and client's network stucked or changed, lotus may cannot publish message to client and chain sync stucked. 

### Logging Information

```text
2021-12-16T01:55:02.655+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2021-12-16T01:55:03.135+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2021-12-16T01:55:03.139+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2021-12-16T01:55:03.140+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2021-12-16T01:55:03.150+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2021-12-16T01:55:03.155+0800    ERROR   chainstore      store/store.go:409      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
```

### Repo Steps

1. Run code on debug mode.
```
func main() {
	ctx := context.Background()
	//token := ""
	url := "ws://127.0.0.1:1234/rpc/v0"
	headers := http.Header{}

	var nodeStruct v0api.FullNodeStruct
	closer, err := jsonrpc.NewMergeClient(context.Background(), url, "Filecoin", []interface{}{&nodeStruct.Internal, &nodeStruct.CommonStruct.Internal}, headers)
	if err != nil {
		panic(err)
	}
	defer closer()
	msg, err := nodeStruct.MpoolSub(ctx)
	if err != nil {
		panic(err)
	}
	for ss := range msg {
		fmt.Printf("%+v\n", ss) // DEBUG ON THIS LINE
	}
}
```
2. wait for a while, lotus chain sync stucked.
3. See error 'reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets'

**Comment by magik6k** [comment]
Can you get a goroutine dump when that happens - `lotus pprof goroutines > goroutines.txt`?

**Comment by ysh0566** [comment]
> Can you get a goroutine dump when that happens - `lotus pprof goroutines > goroutines.txt`?
[goroutines.txt](https://github.com/filecoin-project/lotus/files/7726544/goroutines.txt)

**Comment by Anonyma** [comment]
We are coming across this issue too on v1.14.1

**Comment by Anonyma** [comment]
We have come across this issue too on v1.14.1

**Comment by kevinlv1024** [comment]
I have same error
[goroutines.txt](https://github.com/filecoin-project/lotus/files/8396838/goroutines.txt)
.

**Comment by mur-me** [comment]
Hey, I've gotten this in `1.15.0` version.
Unfortunately, I've restarted lotus and haven't saved goroutines dump.
I haven't found anything special in logs, just a bunch of:
`2022-04-07T09:38:45.634Z	ERROR	chainstore	store/store.go:409	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets`

**Comment by Anonyma** [comment]
Coming back to report that it is still happening on v1.15.1, and actually much more frequently than before.

**Comment by raynaudoe** [comment]
Also reporting that this issue is still present on `v1.15.2`. Nodes are getting out of sync very frequently.

**Comment by RobQuistNL** [comment]
I just ha this happen after a restart of my daemon

**Comment by jleni** [comment]
we have noticed this much more often in the latest versions

**Comment by kakaly** [comment]
+1

**Comment by beck-8** [comment]
> Closing this now as there has been no reports of `reorg channel is heavily backlogged` by node operators on v1.15.3. Node-operators has also seen way less long block duration times after #8635 landed, ref: #8565.
> 
> If someone still experiences this on Lotus v1.15.3 or higher, do not hesitate to reopen this, or open a new issue!

v1.16.0
`reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets`

**Comment by sunjiebin** [comment]
> > Closing this now as there has been no reports of `reorg channel is heavily backlogged` by node operators on v1.15.3. Node-operators has also seen way less long block duration times after #8635 landed, ref: #8565.
> > If someone still experiences this on Lotus v1.15.3 or higher, do not hesitate to reopen this, or open a new issue!
> 
> v1.16.0 `reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets`

+1 , v1.16.0 also has this error, But it disappeared after a few minutes

**Comment by marshyonline** [comment]
+1 on v1.16.0 - I have not seen this issue on any other version until now
[goroutines.txt](https://github.com/filecoin-project/lotus/files/9043031/goroutines.txt)

**Comment by dongjiahong** [comment]
+1  v1.16.0

**Comment by kevinlv1024** [comment]
2022-07-06T14:04:03.764+0800 ERROR chainstore store/store.go:409 reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets

+1  v1.16.0

**Comment by newwebgroup** [comment]
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/100118145/182827520-96d1c796-4b01-40ba-bb1d-d5de6e97a516.png">

+1 v1.16.0

[goroutines (1).txt](https://github.com/filecoin-project/lotus/files/9258947/goroutines.1.txt)

**Comment by ArseniiPetrovich** [comment]
+1 v1.16.0
Is there any workaround? What should we do to fix this issue?

**Comment by magik6k** [comment]
People seeing this since v1.16.0 - are you running PoSt workers?

**Comment by magik6k** [comment]
This is relevant because it would appear that post-worker alive check will just spam heartbeats in a loop with sleep:

https://github.com/filecoin-project/lotus/blob/master/storage/sealer/sched_post.go#L171-L197

Which in some cases, with multiple PoSt workers, because worker communication is over simple http RPCs, may exhaust all default http client connections..

If this is in fact the bug I'll have a fix for it tomorrow

**Comment by RobQuistNL** [comment]
Yes. I haven't seen it in a while, but I've mainly stopped the post worker because of other issues (https://github.com/filecoin-project/lotus/issues/8362)

**Comment by mostcute** [comment]
V1.23.0
```
023-04-26T23:12:12.159+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.165+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.219+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.640+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.650+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.659+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.666+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:12.719+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.140+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.151+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets                            
2023-04-26T23:12:13.160+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.166+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.220+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.641+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.652+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.660+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.666+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:13.720+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.141+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.152+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.160+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.167+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.221+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.642+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets                                                                                            l2cghz3le76lt2bwl6tdftedf2 bafy2bzacecth63frrwmp4oef5jhr52vlzgfyl7j6hglhkvf5km4de4tzdt
2023-04-26T23:12:14.652+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.661+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.667+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:14.721+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.142+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.153+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.161+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.167+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets                                                                                            c2evshvckv27rzsiuh7rkr6nzy bafy2bzaced7nwp77h6o3pdkmicdb4d2idb3l2cghz3le76lt2bwl6tdfte
2023-04-26T23:12:15.222+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.643+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.653+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.662+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.668+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets                                                                                            xc2evshvckv27rzsiuh7rkr6nzy bafy2bzaced7nwp77h6o3pdkmicdb4d2idb3l2cghz3le76lt2bwl6tdft
2023-04-26T23:12:15.722+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:15.901+0800    WARN    sub     sub/incoming.go:588     ignoring indexer message        {"sender": "12D3KooWNTSFywHjGbmGN1aEqJNp54pDKaiwqpthRrvFZETo37pW", "err": "rate limit exceeded"}                                                                c2evshvckv27rzsiuh7rkr6nzy bafy2bzaced7nwp77h6o3pdkmicdb4d2idb3l2cghz3le76lt2bwl6tdfte
2023-04-26T23:12:16.143+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.154+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.163+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.168+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.222+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.644+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.654+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-04-26T23:12:16.663+0800    ERROR   chainstore      store/store.go:419      reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets

```

**Comment by BBQFIL** [comment]
1.24 agent
`2023-12-05T14:39:10.366+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.366+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.434+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.867+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.867+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.867+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.867+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:10.935+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:11.368+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:11.368+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:11.368+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:11.368+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
2023-12-05T14:39:11.435+0800	ERROR	chainstore	store/store.go:419	reorg channel is heavily backlogged, waiting a bit before trying to take process new tipsets
`
